bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/JsonParser.java

```diff
@@ -136,56 +97,39 @@ private void writeToValueJson(Data output) {
         writePerformanceJson(jsonObject.toString());
     }
 
-    private Data populateChildNodes(StackTraceItem stackTraceItem, Data current, String name) {
-        boolean found = false;
+    private Data populateChildNodes(StackTrace stackTraceItem, Data current, String stackTrace) {
         for (Data child : current.children) {
-            if (child.name.equals(name)) {
-                child.value = Math.max(child.value, stackTraceItem.time);
-                current = child;
-                found = true;
-                break;
+            if (child.name.equals(stackTrace)) {
+                return child;
             }
         }
-        if (!found) {
-            Data newChild = new Data(name, stackTraceItem.time, new ArrayList<>());
-            current.children.add(newChild);
-            current = newChild;
-        }
-        return current;
+        Data newChild = new Data(stackTrace, stackTraceItem.time, new ArrayList<>());
+        current.children.add(newChild);
+        return newChild;
     }
 
-    private List<StackTraceItem> populateStackTraceItems(String jsonInput) {
+    private List<StackTrace> populateStackTraceItems(String jsonInput) {
         Gson gson = new Gson();
         JsonArray jsonArr = gson.fromJson(jsonInput, JsonArray.class);
-        List<StackTraceItem> stackTraceItems = new ArrayList<>();
+        List<StackTrace> stackTraceItems = new ArrayList<>();
         for (JsonElement jsonElement : jsonArr) {
-            StackTraceItem person = gson.fromJson(jsonElement, StackTraceItem.class);
+            StackTrace person = gson.fromJson(jsonElement, StackTrace.class);
             stackTraceItems.add(person);
         }
         return stackTraceItems;
     }
 
-    private void removeChildrenByNodeName(Data node, String nodeName) {
-        if (node.name != null && node.name.equals(nodeName)) {
-            node.children.clear();
-            return;
-        }
-        for (Data child : node.children) {
-            removeChildrenByNodeName(child, nodeName);
-        }
-    }
-
     /**
      * This class is used as a custom data class.
      *
      * @since 2201.8.0
      */
-    private static class StackTraceItem {
+    private static class StackTrace {
```

Comment:
Shall we change this class name to something like `StackElement`? Because it is suggested not to have the same name for a field and class name as a good practice.

Reply:
Revert back to StackTraceItem since it contains stacktrace and time both details.